### PR TITLE
fix: try to use next available port

### PR DESCRIPTION
**Pull request by fengmk2**
closes https://github.com/node-modules/detect-port/issues/15

### CODE REVIEW

**Comment by fengmk2** [comment]
cc @gaearon

File: lib/detect-port.js [hunk 1]
```diff
@@ -8,26 +8,40 @@ module.exports = (port, callback) => {
...
-function tryListen(port, callback) {
-  port = parseInt(port) || 0;
+function tryListen(port, maxPort, callback) {
   const server = new net.Server();
 
   server.on('error', err => {
     debug('listen %s error: %s', port, err);
-    port = 0;
+    if (port === 0) {
+      return callback(err);
```
Comment (popomore) [inline lib/detect-port.js]: Why not try another random port?

File: lib/detect-port.js [hunk 2]
```diff
@@ -8,26 +8,40 @@ module.exports = (port, callback) => {
     callback = port;
     port = null;
   }
+  port = parseInt(port) || 0;
```
Comment (xudafeng) [inline lib/detect-port.js]: `parseInt(port, 10)` ？

Comment (fengmk2) [inline lib/detect-port.js]: default radix is 10, no need to set https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/parseInt

File: lib/detect-port.js (hunk 1 above)
Comment (fengmk2) [inline lib/detect-port.js]: random port won't error on normal case. If listen random port error happen, it's hard to try again, maybe "file handle" full on that time.

Comment (popomore) [inline lib/detect-port.js]: This line is not covered

Comment (fengmk2) [inline lib/detect-port.js]: It's hard to mock this error, ignore it for now.

**Comment by fengmk2** [comment]
1.1.1

**Comment by gaearon** [comment]
Thanks!

**Pull request merged**